src/webresource-handler/actions/beginUpload.ts

```diff
@@ -0,0 +1,163 @@
+	}
+
+	const handler = getMultipartUploadHandler();
+	const { fieldName, beginUploadPayload } = parseBeginUpload(request, handler);
+
+	await runFakeDbPatch(request, {
+		[fieldName]: { ...beginUploadPayload, href: 'fake_patch' },
+	});
+
+	const { fileKey, uploadId, uploadParts } =
+		await handler.multipartUpload.begin(fieldName, beginUploadPayload);
+	const uuid = randomUUID();
+	await api.webresource.post({
+		resource: 'multipart_upload',
+		body: {
+			uuid,
+			resource_name: request.resourceName,
+			field_name: fieldName,
+			resource_id: id,
+			upload_id: uploadId,
+			file_key: fileKey,
+			status: 'pending',
+			...beginUploadPayload,
+			expiry_date: Date.now() + 7 * 24 * 60 * 60 * 1000, // 7 days in ms
+			is_created_by__actor: req.user?.actor,
+		},
+		passthrough: {
+			req: permissions.root,
+			tx,
+		},
+	});
+
+	return {
+		body: { [fieldName]: { uuid, uploadParts } },
+		statusCode: 200,
+	};
+};
+
+const parseBeginUpload = (
+	request: ODataActionRequest,
+	webResourceHandler: MultipartUploadHandler,
+) => {
+	const { values } = request;
+	const isValid = validateBeginUpload(values);
+	if (!isValid) {
+		throw new BadRequestError('Invalid begin upload payload');
+	}
+
+	const fieldName = Object.keys(values)[0];
+
+	const webResourceFields = getWebResourceFields(request, false);
+	if (!webResourceFields.includes(fieldName)) {
+		throw new BadRequestError(
+			`The provided field '${fieldName}' is not a valid webresource`,
+		);
+	}
+	const beginUploadPayload = {
+		...values[fieldName],
+		chunk_size:
+			values[fieldName].chunk_size ??
+			webResourceHandler.multipartUpload.getMinimumPartSize(),
+	} satisfies BeginMultipartUploadPayload;
+
+	if (
+		beginUploadPayload.chunk_size <
+		webResourceHandler.multipartUpload.getMinimumPartSize()
+	) {
+		throw new BadRequestError('Chunk size is too small');
+	}
+
+	return {
+		beginUploadPayload,
+		fieldName,
+	};
+};
+
+// We want beginUpload to fail if the initial payload would already break any constraint/hooks/rule
+// this serves as both an optimization and a better client experience as it avoids succeding a beginUpload
+// (which could allow for potentially very large PartUploads) only for it to fail on commitUpload
+// The current approach to try to achieve this behavior creates this fake database patch:
+// It first tries to patch the file metadata (on a separate tx) to something similar to what it would look like
+// and if it breaks any constraint, then it throws and fail to even start the upload
+// Note that because the href is only generated at commitUpload time, we need to fake it for the time being
+// This could potentially be a problem if an application code has a constraint on the href property
+// However, because this is a internal (pine managed) property we assumed that application code won't do so
+const runFakeDbPatch = async (
+	request: ODataActionRequest,
+	fakeDbPatch: FakeWebResourcePatch,
+) => {
+	const fakeTx = await sbvrUtils.db.transaction();
```

Comment:
I am using the transaction this way rather than the more conventional passing the callback as the one where you pass a callback has an automatic handler for re-throwing when the tx is rolled back which is not the behavior we want